File: android-app/src/main/java/com/medziku/plantwatering/data/parts/HumiditySensorImpl.java
package com.medziku.plantwatering.data.parts;

import android.os.Parcel;
import android.os.Parcelable;

import com.medziku.plantwatering.station.parts.HumiditySensor;


public class HumiditySensorImpl extends HumiditySensor implements Parcelable {

    public HumiditySensorImpl(String name, float humidity) {
        super(name, humidity);
    }

    private HumiditySensorImpl(Parcel in) {
        super(in.readString(), in.readFloat());
    }

    public static final Creator<HumiditySensorImpl> CREATOR = new Creator<HumiditySensorImpl>() {
        @Override
        public HumiditySensorImpl createFromParcel(Parcel in) {
            return new HumiditySensorImpl(in);
        }

        @Override
        public HumiditySensorImpl[] newArray(int size) {
            return new HumiditySensorImpl[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(getName());
        dest.writeFloat(getHumidity());
    }
}

File: android-app/src/main/kotlin/com/medziku/plantwatering/data/parts/HumiditySensorImpl.kt
package com.medziku.plantwatering.data.parts

import android.os.Parcel
import android.os.Parcelable

import com.medziku.plantwatering.station.parts.HumiditySensor


class HumiditySensorImpl : HumiditySensor, Parcelable {

    constructor(name: String, humidity: Float) : super(name, humidity)

    private constructor(`in`: Parcel) : super(`in`.readString(), `in`.readFloat())

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeFloat(humidity)
    }

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<HumiditySensorImpl> = object : Parcelable.Creator<HumiditySensorImpl> {
            override fun createFromParcel(`in`: Parcel): HumiditySensorImpl {
                return HumiditySensorImpl(`in`)
            }

            override fun newArray(size: Int): Array<HumiditySensorImpl?> {
                return arrayOfNulls(size)
            }
        }
    }
}
